tests/testing/unit/harness/harness.tsx

```diff
@@ -301,7 +301,11 @@ describe('harness', () => {
 			const h = harness(() => w(MyWidget, {}));
 			h.trigger('*[key="span"]', (node: WNode | VNode) => {
 				if (isVNode(node)) {
-					return node.properties.onclick;
+					const onclick = node.properties.onclick;
```

Comment:
How come we had to change this here?

Reply:
This now needs to check if the event is an object with `handler` or an function

Reply:
this is a test though isn't it, and you haven't changed the test? so why has the vnode changed?

Reply:
I think I now that this is a type guard, and you can't possibly ever get to the handler code - so I think this would be better as just a type cast instead

Reply:
Hmm so the more I think about this I think the more problematic this type change is for users downstream. Given how this is not the common use case, have we considered not changing `VNodeProperties` at all, and instead creating a utility function that basically creates a function with properties decorated on it (thus adhering to the existing interface?). so something like:

```tsx
<button onclick={ eventWithOptions({ capture: true }, () => console.log('onclick')) }>Button</button>
```

or perhaps even nicer if we just want to solve the `passive` case for now like the issue raised:

```tsx
<button onscroll={ passive(() => console.log('onscroll')) } >Button</button>
```

Reply:
Hmm I haven't considered something like that but it would make the typing alot easier

Reply:
Let me look into this

Reply:
For this to work you would need to change the types so that the function could optionally take any function

Reply:
Going back to this problem what your suggesting would still require a change to the event types so we'd still need to check to see if there's a handler here. What could work instead is if we optionally took two values for events?

Reply:
@samends I'm still not sure what the problem with a wrapper function is with regards to types? It looks like you could do something like this without requiring any changes to types (as it works for v7), and then hook into the property on the handler in the vdom to deal with it?

https://codesandbox.io/s/wonderful-sound-kijth?file=/src/main.tsx